### ISSUE TITLE
New Ubuntu PPA with patched build

### ISSUE THREAD

**Issue by savoury1** [ianare/metamorphose2#26]
tl;dr -- Metamorphose2 [PPA](https://launchpad.net/~savoury1/+archive/ubuntu/metamorphose2) for Ubuntu

This is not an issue, rather a log in the record here on what appears to be unmaintained software (especially given various other issues about installing). Finally got around to creating a new PPA with a build that patches #16 (credit to [Alin Andrei](https://launchpad.net/~nilarimogard/+archive/ubuntu/webupd8) for publishing a version of the software with some needed patches already applied).

So for all who are trying to get Metamorphose2 working on a Ubuntu-based system, check the [PPA](https://launchpad.net/~savoury1/+archive/ubuntu/metamorphose2) for a Xenial build that may well work fine with newer series too (perhaps someone can report on that here, as and when they get to trying the new build on a series newer than Xenial). Soon a forked Github copy of the program, that will have all the patches already applied.

**Comment by ianare** [comment]
Yes, it's true, I really don't have the time to maintain this program so much anymore. I appreciate the work you've done on the PPA, thanks :-)

If you have any patches to make the app work better please don't hesitate to send me pull requests.

- ianaré

**Comment by savoury1** [comment]
Thanks @ianaré for your response! Understand you don't have the time to maintain the program much anymore, but appreciate you replying. Have created the pull request (#27) as you suggested, which fixes a critical bug (resulting in the program being essentially non-functional) on various Ubuntu systems including 16.04 Xenial.

Probably quite a "hacky" fix (not yet a Python programmer) but it does work!  :-)

**Comment by savoury1** [comment]
Thanks @ianare for your reply, understand that you don't have time so much to maintain the program anymore but appreciate the response. Have created a pull request (#27) as you suggested, which fixes a critical bug (for various Ubuntu-based systems at least) that essentially stops the program from functioning, including on my Ubuntu 16.04 Xenial based system.

Probably quite "hacky" (not a Python programmer yet!) but it works!  :-)

**Comment by xpelican** [comment]
Reporting as per your request for **Ubuntu 18.04.1**:
'python-imaging' is listed as a dependency, but it has been replaced by 'python-pil'

Do you have any recommendations for installing this properly? Can't wait for the patched version! Thank you for doing this!

UPDATE: I have installed the newer package, Pillow, and tested it on Python2.7 to see if the function calls work the same way. Looks like they do. So I edited the import call in the src/utils.py file to reference Pillow instead, and tried to make another deb to install from.

This time, the problem becomes installing the **python-wxgtk2.8** package, which is proving to be very difficult without breaking packages on the system significantly. I'll try and test further to see if it gets me anywhere, but thought I'd report back to you in case it proves useful in any way.

**Comment by savoury1** [comment]
Thanks for the report, appreciate the feedback for the new PPA package. So today I updated the build to add python-pil as an alternative to the unavailable (in Bionic and newer) python-imaging dependency.

The new Metamorphose2 version is 0.9.0\~beta to reflect that internally (ie. Help/About) it gives this version number (not 0.8.4). So please try installing the 0.9.0\~beta version from the Launchpad PPA and hopefully that will work for you on Bionic now.

**Comment by savoury1** [comment]
Also see https://github.com/savoury1/metamorphose2/releases/tag/v2.0.9.0 for a new source release. The version number (0.9.0 beta) now reflects that reported internally by Metamorphose2, ie. Help/About. A copy of the Launchpad built binary deb file is also now available at the linked release page.
